pydantic/_internal/_dataclasses.py

```diff
@@ -115,7 +115,7 @@ def complete_dataclass(
         )
 
     if types_namespace is None:
-        types_namespace = _typing_extra.get_cls_types_namespace(cls)
+        types_namespace = _typing_extra.merge_cls_and_parent_ns(cls)
```

Comment:
So we could change this to ` _typing_extra.get_module_ns_of(cls).copy()`, but I think we should instead refactor how we handle namespaces for dataclasses in general to be more consistent with that of models.

See https://github.com/pydantic/pydantic/blob/e13b15edbe6cf90c603fbda1246b9156764c810f/pydantic/dataclasses.py#L318-L321